### PR TITLE
Add core/average.py/average_shape to ops/average.py

### PR DESCRIPTION
**Pull request by charlesgauthier-udm**
### Pull Request Checklist:
- [~] This PR addresses an already opened issue (for bug fixes / features)
  - This PR is related to [this issue](https://github.com/ESGF/esgf-cwt/issues/30) outside of clisops
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Documentation has been added / updated (for bug fixes / features)
 - The new ops/average_shape is documented based on the other functions in ops/average.py, Please let me know if anything else needs to be added.
- [X] HISTORY.rst has been updated (with summary of main changes)
- [X] I have added my relevant user information to `AUTHORS.md`

### What kind of change does this PR introduce?: 
This PR exposes average_Shape from clisops/core to clisops/ops

### Does this PR introduce a breaking change?: 
Not to my knowledge

### Other information: 
As mentioned above, this PR is part of the work related to this [issue](https://github.com/ESGF/esgf-cwt/issues/30) outside of clisops

### CODE REVIEW

File: clisops/core/average.py [hunk 1]
```diff
@@ -78,7 +78,7 @@ def average_shape(
     try:
         from xesmf import SpatialAverager
     except ImportError:
-        raise ValueError("Package xesmf >= 0.5.0 is required to use average_shape")
+        raise ValueError("Package xesmf >= 0.6.2 is required to use average_shape")
```
Comment (huard) [inline clisops/core/average.py:81]:
```suggestion
        raise ValueError("Package xesmf is required to use average_shape")
```

Comment (huard) [reply clisops/core/average.py:81]: The required version is set in the pyproject.toml file. 

I've tried to run the test suite with an older xesmf version and had all kinds of problems with dependencies.

File: clisops/ops/average.py [hunk 2]
```diff
@@ -88,6 +89,81 @@ def average_over_dims(
...
+
+        self.params = {"shape": shape, "variable": variable}
+
+        if not shape:
+            raise InvalidParameterValue(
+                "At least one area for averaging must be provided"
+            )
+
+    def _get_file_namer(self):
+        extra = f"_avg-{'shape'}"
```
Comment (huard) [inline clisops/ops/average.py:105]: Why the `{'shape'}`  ?

**Review by huard — APPROVED** [review]
Looks good.

This is unrelated to this PR, but I had some trouble with the installation of the dependencies. toolz and xarray required an upgrade after the conda env update.

File: clisops/ops/average.py (hunk 2 above)
Comment (charlesgauthier-udm) [reply clisops/ops/average.py:105]: it is the simple (temporary?) solution to naming the file, else we would need to either get name information from the polygon file itself, or ask the user to name their polygon. I can implement both if deemed necessary.

Comment (huard) [reply clisops/ops/average.py:105]: But then it would just be `extra="_avg-shape"` ?

Comment (charlesgauthier-udm) [reply clisops/ops/average.py:105]: Ah, correct yes I'll make the change.

File: HISTORY.rst [hunk 3]
```diff
@@ -13,6 +13,10 @@ Other Changes
 * Warnings are now emitted if the user attempts to run the regridding utilities with a version of `xarray` that is not compatible with `cf-xarray`. (#310).
 * Dependency pins now constrain the `xarray` version when installing with `$ pip install ".[extra]"`. (#310).
 
+New Features
+^^^^^^^^^^^^
+* ``clisops.ops.average.average_shape`` added (#XXX). Exposing average_shape from clisops.core to clisops.ops.
+
```
Comment (Zeitsperre) [inline HISTORY.rst:19]: @charlesgauthier-udm This should be placed in a new section for `v0.12.2 (unreleased)`

Comment (charlesgauthier-udm) [reply HISTORY.rst:19]: Will do and will add PR number as well

**Review by Zeitsperre — APPROVED** [review]

**Comment by Zeitsperre** [comment]
Congrats @charlesgauthier-udm !

I'll stage a new release, given that we now have some new functions from both you and Martin (@sol1105).

**Pull request merged**